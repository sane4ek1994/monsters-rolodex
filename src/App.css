body {
  max-width: 1200px;
  margin: 0 auto;
  padding: 40px 0;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen', 'Ubuntu', 'Cantarell', 'Fira Sans',
    'Droid Sans', 'Helvetica Neue', sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  background: linear-gradient(to left, rgba(7, 27, 82, 1) 0%, rgba(0, 128, 128, 1) 100%);
  text-align: center;
}

.app-title {
  font-size: 76px;
  color: #0ccac4;
  font-family: 'Bigelow Rules';
}
