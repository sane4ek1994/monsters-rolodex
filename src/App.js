import { Component } from 'react'

import { CardList, SearchBox } from './components'

import './App.css'

class App extends Component {
  constructor() {
    super()
    this.state = {
      monsters: [],
      searchString: ''
    }
  }

  onSearchChange = event => {
    const searchString = event.target.value.toLocaleLowerCase()

    this.setState(() => {
      return { searchString }
    })
  }

  componentDidMount() {
    fetch('https://jsonplaceholder.typicode.com/users')
      .then(json => json.json())
      .then(users => this.setState(() => ({ monsters: users })))
  }

  render() {
    const { monsters, searchString } = this.state
    const { onSearchChange } = this

    const filteredMonsters = monsters.filter(monster => monster.name.toLocaleLowerCase().includes(searchString))

    return (
      <div className='App'>
        <h1 className='app-title'>Monsters Rolodex</h1>
        <SearchBox onChangeHandler={onSearchChange} placeholder='search monsters' className='search-box' />
        <CardList monsters={filteredMonsters} className='card-list' />
      </div>
    )
  }
}

export default App
