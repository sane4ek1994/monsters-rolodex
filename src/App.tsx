import React from 'react'

import { CardList, SearchBox } from './components'
import { getData } from './utils/data.utils'

import './App.css'

export type Monster = {
  id: string
  name: string
  email: string
}

const App = () => {
  const [searchField, setSearchField] = React.useState('')
  const [title, setTitle] = React.useState('')
  const [monsters, setMonsters] = React.useState<Monster[]>([])
  const [filteredMonsters, setFilteredMonsters] = React.useState(monsters)

  React.useEffect(() => {
    const fetchUsers = async () => {
      const users = await getData<Monster[]>('https://jsonplaceholder.typicode.com/users')
      setMonsters(users)
    }

    fetchUsers()
  }, [])

  React.useEffect(() => {
    const newFilteredMonsters = monsters.filter(monster => monster.name.toLocaleLowerCase().includes(searchField))
    setFilteredMonsters(newFilteredMonsters)
  }, [monsters, searchField])

  const onTitleChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
    const searchFieldString = event.target.value.toLocaleLowerCase()
    setTitle(searchFieldString)
  }

  const onSearchChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
    const searchFieldString = event.target.value.toLocaleLowerCase()
    setSearchField(searchFieldString)
  }

  return (
    <div className='App'>
      <h1 className='app-title'>{title}</h1>
      <SearchBox onChangeHandler={onSearchChange} placeholder='search monsters' className='search-box' />
      <br />
      <SearchBox onChangeHandler={onTitleChange} placeholder='title' className='search-box' />
      <CardList monsters={filteredMonsters} />
    </div>
  )
}

export default App
