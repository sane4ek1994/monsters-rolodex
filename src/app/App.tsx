import React from 'react'

import { CardList, SearchBox } from '../components'
import { getData } from '../utils/data.utils'

import './App.css'

export type Monster = {
  id: string
  name: string
  email: string
}

const App = () => {
  const [counter, setCounter] = React.useState(0)
  const [searchField, setSearchField] = React.useState('')
  const [title, setTitle] = React.useState('Monsters Rolodex')
  const [monsters, setMonsters] = React.useState<Monster[]>([])
  const [filteredMonsters, setFilteredMonsters] = React.useState(monsters)

  React.useEffect(() => {
    const fetchUsers = async () => {
      const users = await getData<Monster[]>('https://jsonplaceholder.typicode.com/users')
      setMonsters(users)
    }

    fetchUsers()
  }, [])

  React.useEffect(() => {
    const newFilteredMonsters = monsters.filter(monster => monster.name.toLocaleLowerCase().includes(searchField))
    setFilteredMonsters(newFilteredMonsters)
  }, [monsters, searchField])

  const incCounter = () => setCounter(counter => counter + 1)

  const onTitleChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
    const searchFieldString = event.target.value.toLocaleLowerCase()
    setTitle(searchFieldString)
  }

  const onSearchChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
    const searchFieldString = event.target.value.toLocaleLowerCase()
    setSearchField(searchFieldString)
  }

  return (
    <div className='App'>
      <button onClick={incCounter} id='counter'>
        counter++{counter}
      </button>
      <h1 className='app-title'>{title}</h1>
      <SearchBox onChangeHandler={onSearchChange} placeholder='search monsters' className='search-box' />
      <br />
      <SearchBox onChangeHandler={onTitleChange} placeholder='title' className='search-box' />
      <CardList monsters={filteredMonsters} />
    </div>
  )
}

export default App
