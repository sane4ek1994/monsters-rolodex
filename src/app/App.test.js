import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import App from './App'

it('expect to render app component', () => {
  const app = shallow(<App />)
  expect(toJson(app)).toMatchSnapshot()
})

it('corretcly increments the counter', () => {
  const wrapper = shallow(<App />)
  wrapper.find('[id="counter"]').simulate('click')
  wrapper.find('[id="counter"]').simulate('click')
  expect(wrapper.state()).toEqual({ counter: 2 })
})
