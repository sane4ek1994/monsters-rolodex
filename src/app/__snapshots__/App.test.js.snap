// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render app component 1`] = `
<div
  className="App"
>
  <button
    id="counter"
    onClick={[Function]}
  >
    counter++
    0
  </button>
  <h1
    className="app-title"
  >
    Monsters Rolodex
  </h1>
  <SearchBox
    className="search-box"
    onChangeHandler={[Function]}
    placeholder="search monsters"
  />
  <br />
  <SearchBox
    className="search-box"
    onChangeHandler={[Function]}
    placeholder="title"
  />
  <CardList
    monsters={Array []}
  />
</div>
`;
